packages/components/src/MultiSelect/subcomponents/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useRef, InputHTMLAttributes } from "react"
+import classnames from "classnames"
+import { CheckIcon, MinusIcon } from "~components/Icons"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./Checkbox.module.scss"
+
+export type CheckedStatus = "checked" | "unchecked" | "indeterminate"
+
+type ReadOnly = {
+  readOnly: true
+  onChange?: never
+}
+
+type Controlled = {
+  readOnly?: false
```

Comment:
This is feeling quite influenced by what the old one was doing. Do you think you would do it this way if you were looking at creating a fresh Checkbox component?

Reply:
It was also so that for our stories we wouldn't need to write `onChange={() => undefined` constantly, given we want it to be required.
Doing this fresh, we would probably not have considered `readOnly` until we had designs for it, but knowing about it just made it a little nicer for the consuming api.